testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/mail/basic/MailTesterServlet.java

```diff
@@ -47,27 +47,42 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
             message.setContent(emailContent, "text/plain");
             Transport.send(message);
 
-            // Wait a bit to ensure the server has processed the message
-            TimeUnit.SECONDS.sleep(1);
+            long endTime = System.currentTimeMillis() + TimeoutUtil.adjust(5000);
+            do {
+                Thread.sleep(500);
+                try {
+                    tryToReadEmails(emailContent);
+                    return;
+                } catch (MessagingException | IOException e){
+                    throw new ServletException(e);
+                } catch (ServletException e){
+                    if (e.getMessage().equals("Message read is not equals to the message sent")){
+                        throw e;
+                    }
+                }
```

Comment:
Looks ok, but as a suggestion to make it more simpler:

1. tryToReadEmails could just return true/false. With that method, we are now interested only into know whether the emails were read as we expected. I mean, if `messages.length != 0 and the message text is the expected one` return true, otherwise false.
2. Any MessagingException or IOException can be ignored by the caller, and only go out of the loop if the method returned true.

it is up to you @Dkafetzis , I am fine as it is if the test case passes, but simplifying it a bit more would be nice.

Reply:
@yersan I agree that we can ignore the MessagingException and the IOException, but I'd like to keep some way to know if the received message is different from what is expected and not just keep retrying till timeout in this case, since this indicates a different problem and I think it would be helpful to know if it fails because of this and not just due to the mail server not being up and running yet.

Reply:
Thanks @Dkafetzis , I'm more in favor of simplicity, but that's ok for me for this test case.

Reply:
@yersan Great, I applied the changes as requested then. Should be ready now